server/data/data-api/src/main/java/org/apache/james/user/api/UsersRepository.java

```diff
@@ -142,4 +142,10 @@
      */
     boolean isAdministrator(String username) throws UsersRepositoryException;
 
+    /**
+     * @return true if one can use {@link UsersRepository#updateUser(User)} {@link UsersRepository#addUser(String, String)}
```

Comment:
I don't think we should pay it right now. The original idea of spltting interfaces seems fine, but very complex from an injection perspective. And it really breaks the actual implementation.
A good solution would be to split the repository in two different things: an authentication service, which by nature is readonly (let's say that to simplify). And a real userRepository, which can be modified.